vaadin-radio-button-flow-demo/src/main/java/com/vaadin/flow/component/radiobutton/demo/RadioButtonGroupView.java

```diff
@@ -15,330 +15,287 @@
         // end-source-example
 
-        group.setId("button-group-with-value-change-listener");
-        message.setId("button-group-value");
-
-        addCard("Basic radio button group", group, message);
+        disabledRadioGroup.getStyle().set("margin-right", "7.5em");
+        VerticalLayout verticalLayout = new VerticalLayout(disabledRadioGroup,
+                radioGroup);
+        addCard("Disabled state", verticalLayout);
     }
 
-    private void addItemRenderer() {
-        Div message = new Div();
-
+    private void readonlyRadioButton() {
         // begin-source-example
-        // source-example-heading: Radio button group with renderer
-        RadioButtonGroup<Person> group = new RadioButtonGroup<>();
-        group.setItems(new Person(1, "Joe"), new Person(2, "John"),
-                new Person(3, "Bill"));
-        group.setRenderer(new ComponentRenderer<>(person -> new Anchor(
-                "http://example.com/" + person.getId(), person.getName())));
-        group.addValueChangeListener(event -> message.setText(String.format(
-                "Radio button group value changed from '%s' to '%s'",
-                getName(event.getOldValue()), getName(event.getValue()))));
+        // source-example-heading: Read-only
+        RadioButtonGroup<String> radioGroup = new RadioButtonGroup<>();
+        radioGroup.setItems("Option one", "Option two", "Option three");
+        radioGroup.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+        radioGroup.setValue("Option one");
+        radioGroup.setReadOnly(true);
         // end-source-example
 
-        group.setId("button-group-renderer");
-        message.setId("button-group-renderer-value");
-
-        addCard("Radio button group with renderer", group, message);
+        addCard("Read-only", radioGroup);
     }
 
-    private void addItemLabelGenerator() {
-        Div message = new Div();
-
-        // begin-source-example
-        // source-example-heading: Radio button group with label generator
-        RadioButtonGroup<Person> group = new RadioButtonGroup<>();
-        group.setItems(new Person("Joe"), new Person("John"),
-                new Person("Bill"));
-        group.setRenderer(new TextRenderer<>(Person::getName));
-        group.addValueChangeListener(event -> message.setText(String.format(
-                "Radio button group value changed from '%s' to '%s'",
-                getName(event.getOldValue()), getName(event.getValue()))));
-        // end-source-example
-
-        group.setId("button-group-with-item-generator");
-        message.setId("button-group-gen-value");
+    private List<Department> getDepartments() {
 
-        addCard("Radio button group with label generator", group, message);
+        DepartmentData departmentData = new DepartmentData();
+        return departmentData.getDepartments();
     }
 
-    private void addItemIconGenerator() {
+    private void entityList() {
```

Comment:
Let's move this as the second demo (right after the basic usage demo) since this is more of a realistic use-case

Reply:
Done.